Dynatrace/dynatrace-operator
# Refactor ProcessmoduleConfig code and add missing unit test

## Description

This is a refactor of some parts of the `ProcessModuleConfig` handling. Namely: the `OneAgentProvisioner` sets some values which are currently called every time per `CSI Driver`. Now that they are moved to the `Dynakube` reconciler, they are called on `Dynakube` reconcile. In order to make sure that the `OneAgentTenantToken` is ready at the time when the `ProcessModuleConfig` is getting reconciled. This is because due to the refactor there `OneAgentConnectionInfo` info is added to the `LatestProcessModuleConfig`.

Also: I did some mini refactor in c1e69c33ab31f2fde456bb1a6a2f1fc9c405d4e7

## How can this be tested?

Unit tests

and/or

Deploy operator, inject into namespace and check dynakube-pmc-secret 

## Checklist
